src/metabase/api/permissions.clj

```diff
@@ -46,6 +48,16 @@
                            (spec/explain-str ::pg/data-permissions-graph body))
                       {:status-code 400
                        :error       (spec/explain-data ::pg/data-permissions-graph body)})))
+
+    (let [has-block-perms? (atom false)]
+      (walk/postwalk (fn [x]
+                       (if (and (map? x) (= (:schemas x) :block))
+                         (do (reset! has-block-perms? true)
+                             nil)
+                         x)) graph)
+      (when (and @has-block-perms? (not (premium-features/has-feature? :advanced-permissions)))
+        (throw (ex-info (tru "Can''t use block permissions without having the advanced-permissions premium feature")
+                        {:status-code 403}))))
```

Comment:
I've been using [`402 Payment Required`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/402) for the other feature-flagged responses. Not 100% sure it makes sense but it does at least make the tests easier to write because we can confirm something is failing because of token feature flags rather than perms

Reply:
In my impl for the fix I just added this check to `update-db-permissions!` in the `:block` code... that way you don't have to do any walking

Reply:
Fixed both